### ISSUE TITLE
Crash when trying to show a table between pages

### ISSUE THREAD

**Issue by cetorres** [DavBfr/dart_pdf#57]
**Describe the bug**
It's crashing when trying to show a table between pages. Sometimes when there is a certain content in the page and the page ends with a Table, it crashes when trying to add the Table to the page with the following error in the Screenshot.

**To Reproduce**
I'm just adding content to a build MultiPage widget normally. Adding Paragraphs, Tables etc directly.

**Expected behavior**
Calculate correctly the table heights to avoid the error.

**Screenshots**
<img width="951" alt="Screenshot" src="https://user-images.githubusercontent.com/533661/55972322-4a4a4b00-5c51-11e9-8466-c285846884de.png">

**Stacktrace**
```
E/flutter (19441): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: RangeError (end): Invalid value: Only valid value is 0: -1
E/flutter (19441): #0      RangeError.checkValidRange (dart:core/errors.dart:333)
E/flutter (19441): #1      _GrowableList.sublist (dart:core/runtime/libgrowable_array.dart:81)
E/flutter (19441): #2      TableBorder.paintBorders (file:///Users/cetorres/.pub-cache/hosted/pub.dartlang.org/pdf-1.3.7/lib/widgets/table.dart:73)
E/flutter (19441): #3      Table.paint (file:///Users/cetorres/.pub-cache/hosted/pub.dartlang.org/pdf-1.3.7/lib/widgets/table.dart:301)
E/flutter (19441): #4      MultiPage._paintChild (file:///Users/cetorres/.pub-cache/hosted/pub.dartlang.org/pdf-1.3.7/lib/widgets/multi_page.dart:78)
E/flutter (19441): #5      MultiPage.generate (file:///Users/cetorres/.pub-cache/hosted/pub.dartlang.org/pdf-1.3.7/lib/widgets/multi_page.dart:198)
E/flutter (19441): #6      Document.addPage (file:///Users/cetorres/.pub-cache/hosted/pub.dartlang.org/pdf-1.3.7/lib/widgets/document.dart:47)
E/flutter (19441): #7      CreatePDF.reportForCustomer (package:imt_survey/utils/create_pdf.dart:60)
```

**Comment by cetorres** [comment]
I was testing doing some changes in your code and if I do a test for the heights list length > 0 it doesn't crash anymore, but the result gets a little messy.

```dart
   if (horizontalInside && heights.length > 0) { // Added heights.length > 0
      double offset = box.top;
      for (double height in heights.sublist(0, heights.length - 1)) {
        offset -= height;
        context.canvas.moveTo(box.x, offset);
        context.canvas.lineTo(box.right, offset);
      }
      context.canvas.strokePath();
    }
```

You can see the result below. See the table in the end of the first page, beginning of second page.

<img width="764" alt="Screen Shot 2019-04-11 at 4 55 34 PM" src="https://user-images.githubusercontent.com/533661/55992603-f3f20200-5c7a-11e9-9fc6-2bd2f4e3f725.png">

**Comment by DavBfr** [comment]
Can you give me an example that crashes? I can't reproduce.

**Comment by cetorres** [comment]
I know, it's not easy to reproduce. The same code produces the crash, depending on the data that I'm using, that influences the creation of more tables or pushes a table in a way in the page that it cannot fit well and crashes.

**Comment by DavBfr** [comment]
Fixed in pdf 1.3.10
